### PR TITLE
Added support for querying attributes by slug

### PR DESCRIPTION
**Pull request by LeOndaz**
demonstration
```
query attrBySlug() {
  attribute(slug: "brand"){
    id
    name
    values{
      name
    }
  }
}
```

### CODE REVIEW

**Comment by maarcingebala** [comment]
Thanks for opening the PR! The code looks good but we need two more things:
- GraphQL schema needs to be updated and committed, you can update it locally with `npm run build-schema`.
- Add a test for fetching an attribute by slug.

File: saleor/graphql/product/schema.py [hunk 1]
```diff
@@ -231,8 +231,12 @@ class ProductQueries(graphene.ObjectType):
     def resolve_attributes(self, info, **kwargs):
         return resolve_attributes(info, **kwargs)
 
-    def resolve_attribute(self, info, id):
-        return graphene.Node.get_node_from_global_id(info, id, Attribute)
+    def resolve_attribute(self, info, id=None, slug=None):
+        validate_one_of_args_is_in_query("id", id, "slug", slug)
+        if id:
+            return graphene.Node.get_node_from_global_id(info, id, Product)
+        if slug:
```
Comment (korycins) [inline saleor/graphql/product/schema.py:238]: IMO the second if is no needed. You validated the args in 235, which means that you have id or slug.
```
if id:
    return graphene.Node.get_node_from_global_id(info, id, Product)
return resolve_attribute_by_slug(slug=slug)
```

Comment (maarcingebala) [reply saleor/graphql/product/schema.py:238]: @korycins You're right, but this solution is consistent with our codebase, we have the same structure if few more resolvers. So I think this is fine for this PR.

Comment (LeOndaz) [reply saleor/graphql/product/schema.py:238]: > IMO the second if is no needed. You validated the args in 235, which means that you have id or slug.
> 
> ```
> if id:
>     return graphene.Node.get_node_from_global_id(info, id, Product)
> return resolve_attribute_by_slug(slug=slug)
> ```

I borrowed this from the other resolvers for consistency

Comment (LeOndaz) [reply saleor/graphql/product/schema.py:238]: I'll make it like this
```
if id:
    return graphene.Node.get_node_from_global_id(info, id, Product)
return resolve_attribute_by_slug(slug=slug)
```
but for all the codebase, soon

Comment (korycins) [reply saleor/graphql/product/schema.py:238]: 👍 


> @korycins You're right, but this solution is consistent with our codebase, we have the same structure if few more resolvers. So I think this is fine for this PR.

Sounds good 👍

Comment (LeOndaz) [reply saleor/graphql/product/schema.py:238]: @korycins, @maarcingebala Can anyone review the current commit?

**Review by korycins — APPROVED** [review]

**Comment by LeOndaz** [comment]
May I know the what will happen next?

**Comment by korycins** [comment]
@LeOndaz We are waiting for @maarcingebala 's review

**Comment by LeOndaz** [comment]
> @LeOndaz We are waiting for @maarcingebala 's review

Okay, thanks

**Comment by maarcingebala** [comment]
@LeOndaz Looks good! The PR needs to be rebased with master as there are some conflicts, probably caused by multichannel or page types that we merged this week.

**Review by maarcingebala — APPROVED** [review]

**Comment by LeOndaz** [comment]
I commited by mistake, don't waste time revising this, I'll also add another feature after this as there's something essential missing,

**Comment by LeOndaz** [comment]
I reversed the mistaken commit, I've a question, why the attributes queries where removed from ProductQueries? as this is the main conflict

**Comment by maarcingebala** [comment]
Yes, I should have mentioned that in my previous comment - attributes queries were moved to the `graphql/attribute/schema.py` file. The attributes logic is growing, as now they can be used not only with products but also with pages. To make the logic more universal and easier to maintain, we moved attributes to their own graphql module. You should move your changes from `saleor/graphql/product` to `saleor/graphql/attribute`.

**Comment by LeOndaz** [comment]
Did I just run black and isort on the whole project instead of just the files I edited.. hmmm, anyway, this should be working, Review and if there're no issues and you're okay with it accept asap as I want to make another PR

**Comment by maarcingebala** [comment]
These linters are still failing: `isort`, `black` and `flake8`. Make sure to use exact versions of linters that are defined in `pyproject.toml` and run them against the entire codebase.

**Comment by LeOndaz** [comment]
I was busy the last week so I couldn't do this, anyway, everything is up and running and every test should be passed soon,. I'll be waiting for the merge and I'll be creating another PR soon.

**Review by maarcingebala — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Comment by LeOndaz** [comment]
@korycins @maarcingebala I have 2 questions if I may, I have some intentions to PR a big update, adding reviews to Saleor system (I've already implemented most of this), your're okay with me adding it? (Asking because II know it's on your future plans). Secondly, what is the product rating included in Saleor by default? if it's just a normal rating, what's the purpose of it being a normal number (not from 1-5?)

**Comment by patrys** [comment]
@LeOndaz When such large chunks of functionality are at stake, we will need to work on the design first (both UI and API). I suggest approaching the team before working on a particular implementation, so there's time to discuss the approach (e.g., what ranged to use and whether we need to define the ranges at all). A thread in GitHub discussions may be a good place to gather feedback.

For this particular PR, we'll have to wait for a bit. Our legal department recommended that we implement a CLA process, but we're still waiting for the actual CLA text for contributors to agree to.

**Comment by LeOndaz** [comment]
@patrys I understand, it was easier for me to PR this than creating a plugin to handle it because graphene federation doesn't work in saleor (because of graphene v2 and not saleor), anyway thanks for replying. I'd like to know what's the product rating and why it is implemented like this (a number from 0 to whatever)

**Pull request merged**